rotkehlchen/chain/evm/decoding/cowswap/decoder.py

```diff
@@ -369,7 +370,20 @@ def _decode_normal_claim(self, context: DecoderContext) -> DecodingOutput:
         raw_amount = hex_or_bytes_to_int(context.tx_log.data[128:160])
         amount = asset_normalized_value(amount=raw_amount, asset=self.vcow_token)
         airdrop_identifier: Literal['cow_mainnet', 'cow_gnosis'] = 'cow_mainnet' if self.evm_inquirer.chain_id == ChainID.ETHEREUM else 'cow_gnosis'  # noqa: E501
+        # claimTypes with payment: 1=GnoOption, 2=UserOption, 3=Investor
+        claim_has_payment = hex_or_bytes_to_int(context.tx_log.data[32:64]) in (1, 2, 3)
+        claimant_address = hex_or_bytes_to_address(context.tx_log.data[64:96])
         for event in context.decoded_events:
+            # Claim event always follows payment. Continue on payment, break on claim.
+            if (
+                claim_has_payment and
+                event.location_label == claimant_address and
+                event.event_type == HistoryEventType.SPEND and
+                event.counterparty != CPT_GAS
```

Comment:
Shouldn't you check for the asset here? Excluding gas seems arbitrary .

The assets with which you could pay for this are definitely limited

Reply:
Yes missed this. Can be only ETH, XDAI or GNO unless I'm mistaken